psd-web/app/forms/two_factor_authentication_form.rb

```diff
@@ -0,0 +1,22 @@
+class TwoFactorAuthenticationForm
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+
+  # From rails validations guide: https://guides.rubyonrails.org/active_record_validations.html#numericality
+  NUMERICALITY_REGEX = /\A[+-]?\d+\z/.freeze
```

Comment:
I think we only need to check for digits, not `+` or `-` prefixes, as we’re not _really_ checking whether it’s a logical number or not.

```suggestion
  SECURITY_CODE_REGEX = /\A\d+\z/.freeze
```